docs/learn/expert-guides/universal-profile/extend-profile-functionalities.md

```diff
@@ -0,0 +1,556 @@
+sidebar_position: 4
+---
+
+# Extend a Universal Profile Functionalities
+
+:::info Requirements
+
+You will need a Universal Profile that you can control via its KeyManager to follow this guide. <br/>
+:arrow_left: If you don't have a Universal Profile yet, follow our previous guide [**Create a Universal Profile**](./create-profile.md).
+
+:::
+
+This guide will teach you how to **add functionalities to your Universal Profile** programmatically in JavaScript and includes:
+
+- deploying extensions to the profile
+- extending the Universal Profile with extensions
+- extending the Universal Profile interfaceIds
+
+To achieve this goal, we will perform the following steps:
+
+1. Deploy contracts that have the logic to be added to the Universal Profile
+2. Attach the extension addresses to the Universal Profile
+
+## Introduction
+
+:::info
+
+Read the standard docs about including [LSP17-ContractExtension in the Universal Profile](../../../standards/universal-profile/lsp0-erc725account.md#lsp17---contract-extension).
+
+:::
+
+A Universal Profile contains non-biased features, including generic execution capabilities, the ability to attach generic information, and the ability to be notified and react to various actions. However, post-deployment, a Universal Profile, like any smart contract, becomes immutable. This immutability restricts the ability to add new functionalities into the code of the contract.
+
+To overcome this limitation, the concept of **extensions** is introduced. Extensions are essentially external contracts that act as supplementary modules to the Universal Profile. They provide a dynamic way to enhance the profile's capabilities.
+
+When a call is made to the Universal Profile for a function that isn't natively part of its primary code, the contract checks if this function exists as an extension. If found, the call is seamlessly redirected to the appropriate external contract (extension), thereby effectively expanding the functionality of the Universal Profile without altering its core, immutable code. This approach enables greater flexibility and adaptability, ensuring the Universal Profile can evolve to meet diverse requirements.
+
+## Extending Functionalities
+
+### Setup
+
+Set up a new project with:
+
+```bash
+npm init
+```
+
+Then, install the dependencies
+
+```bash
+npm install ethers@v5 @lukso/lsp-smart-contracts@v0.14
+```
+
+### Step 1: Instantiate the Universal Profile
+
+#### Generate a Signer
+
+First, we need to generate a signer. This signer will be used to interact with the blockchain. We'll use a private key of the Universal Profile controller and an RPC URL.
+
+```js
+const { ethers } = require('ethers');
+
+// RPC URL (e.g., LUKSO testnet)
+const rpcUrl = 'https://rpc.testnet.lukso.network';
+
+// Replace with your private key
+const privateKey = 'your-private-key';
+
+const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+const signer = new ethers.Wallet(privateKey, provider);
+```
```

Comment:
just putting this here: https://docs.lukso.tech/learn/smart-contract-developers/getting-started#install-packages--setup-tools

and make sure there is no duplicate with this other file.

In toher guides we say: check the getting started to have everything ready then come back here.

Reply:
But this page is not related and doesn't provide the code that I have as a setup except the `@lsp-smart-contracts` package, so I don't think its worth referencing a page for this package, also will create some confusion with hardhat